### ISSUE TITLE
Advocate stops working with requests

### ISSUE THREAD

**Issue by mercuree** [JordanMilne/Advocate#8]
Since requests 2.14.0 advocate is broken. 
requests updated its internal urllib3 version and this can be the cause of the problem.
I get an error which relates to urllib3:
```
File "D:\******\******\env36\lib\site-packages\requests\sessions.py", line 531, in get
    return self.request('GET', url, **kwargs)
  File "D:\******\******\env36\lib\site-packages\requests\sessions.py", line 518, in request
    resp = self.send(prep, **send_kwargs)
  File "D:\******\******\env36\lib\site-packages\requests\sessions.py", line 639, in send
    r = adapter.send(request, **kwargs)
  File "D:\******\******\env36\lib\site-packages\requests\adapters.py", line 403, in send
    conn = self.get_connection(request.url, proxies)
  File "D:\******\******\env36\lib\site-packages\requests\adapters.py", line 307, in get_connection
    conn = self.poolmanager.connection_from_url(url)
  File "D:\******\******\env36\lib\site-packages\requests\packages\urllib3\poolmanager.py", line 279, in connection_from_url
    pool_kwargs=pool_kwargs)
  File "D:\******\******\env36\lib\site-packages\requests\packages\urllib3\poolmanager.py", line 227, in connection_from_host
    return self.connection_from_context(request_context)
  File "D:\******\******\env36\lib\site-packages\requests\packages\urllib3\poolmanager.py", line 238, in connection_from_context
    pool_key = pool_key_constructor(request_context)
  File "D:\******\******\env36\lib\site-packages\requests\packages\urllib3\poolmanager.py", line 103, in _default_key_normalizer
    return key_class(**context)
TypeError: __new__() got an unexpected keyword argument 'key_validator'
```

**Comment by KevinHock** [comment]
I got this too.

**Comment by JordanMilne** [comment]
I'll be working on this this weekend. The underlying APIs and testing methods for requests have changed enough that I'm going to drop backwards compat for requests earlier than 2.13 (maybe 2.18.)

**Comment by firba1** [comment]
I also ran into this.

**Comment by barakt11** [comment]
any update here?

**Comment by KevinHock** [comment]
Hi @marataziat, with all due respect I wouldn't suggest using this on production at this time. (I wouldn't use [PyT](https://github.com/python-security/pyt) on production either ;) )

**Comment by marataziat** [comment]
KevinHock ok :3

**Comment by JordanMilne** [comment]
This should have been fixed by advocate 1.0.0, requests 2.18.0 and above are now supported. As a side note, I agree with @KevinHock. Plain requests with a properly configured egress proxy is far preferable :)
